plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/part/AssociationPart.java

```diff
@@ -226,34 +231,15 @@ public void setSelected(int value) {
             return;
         }
 
-        markAssociatedAttributes(value);
-    }
-
-    public void markAssociatedAttributes(int value) {
-        //Color columnColor = value != EditPart.SELECTED_NONE ? Display.getDefault().getSystemColor(SWT.COLOR_RED) : getViewer().getControl().getForeground();
-        //boolean isSelected = value != EditPart.SELECTED_NONE;
-        if (getSource() instanceof EntityPart) {
-            for (AttributePart attrPart : getEntityAttributes((EntityPart) getSource(), getAssociation().getSourceAttributes())) {
-                //attrPart.getFigure().setForegroundColor(columnColor);
-                attrPart.setSelected(value);
-            }
-        }
-        if (getTarget() instanceof EntityPart) {
-            for (AttributePart attrPart : getEntityAttributes((EntityPart) getTarget(), getAssociation().getTargetAttributes())) {
-                //attrPart.getFigure().setForegroundColor(columnColor);
-                attrPart.setSelected(value);
-            }
-        }
-    }
-
-    private List<AttributePart> getEntityAttributes(EntityPart source, List<ERDEntityAttribute> columns) {
-        List<AttributePart> result = new ArrayList<>();
-        for (Object attrPart : source.getChildren()) {
-            if (attrPart instanceof AttributePart && columns.contains(((AttributePart)attrPart).getAttribute())) {
-                result.add((AttributePart)attrPart);
+        if (value != EditPart.SELECTED_NONE) {
+            if (this.getViewer() instanceof ERDGraphicalViewer && associatedAttributesHighlighing == null) {
```

Comment:
These `if` statements can be merged.

Reply:
These two conditions should not be merged, it'll break the logic.